src/libexec/zmsearchndelete

```diff
@@ -0,0 +1,946 @@
+    my $resp = do_delegateauth(@_);
+    my $soap = $opt->{soap};
+    my $hdr  = auth_info( resp => $resp );
+
+    # use call to get sessionId will also identify bad accounts...
+    $resp = do_getinfo( { soap => $soap, head => $hdr->{header} }, "mbox" );
+
+    return auth_info( resp => $resp, token => $hdr->{token} );
+}
+
+# do_adminauth( { soap => $soap }, $Opt{authuser}, $Opt{password} );
+sub do_adminauth {
+    my $opt = ( ref( $_[0] ) eq "HASH" ) ? shift : {};
+    my ( $name, $password ) = @_;
+
+    my $head = SOAP::Header->name("context")->attr( { xmlns => NS_ZADMIN } );
+    my $body = [
+        SOAP::Data->name( name     => $name ),
+        SOAP::Data->name( password => $password ),
+    ];
+
+    return call_soap(
+        %$opt,
+        req  => "AuthRequest",
+        attr => { xmlns => NS_ZADMIN },
+        head => $head,
+        body => $body
+    );
+}
+
+# do_auth( { soap => $soap }, $Opt{account}, $Opt{password} );
+sub do_auth {
+    my $opt = ( ref( $_[0] ) eq "HASH" ) ? shift : {};
+    my ( $acct, $password ) = @_;
+
+    my $head = SOAP::Header->name("context")->attr( { xmlns => NS_ZIMBRA } );
+    my $body = [
+        SOAP::Data->name( account  => $acct )->attr( { by => "name" } ),
+        SOAP::Data->name( password => $password ),
+    ];
+
+    return call_soap(
+        %$opt,
+        req  => "AuthRequest",
+        attr => { xmlns => NS_ZACCOUNT },
+        head => $head,
+        body => $body
+    );
+}
+
+# do_delegateauth( { soap => $soap }, "id|name" => $acct );
+sub do_delegateauth {
+    my $opt = ( ref( $_[0] ) eq "HASH" ) ? shift : {};
+    my ( $by, $acct ) = @_;
+
+    my $body = [ SOAP::Data->name( account => $acct )->attr( { by => $by } ), ];
+
+    return call_soap(
+        %$opt,
+        req  => "DelegateAuthRequest",
+        attr => { xmlns => NS_ZADMIN },
+        body => $body
+    );
+}
+
+# do_getinfo( { soap => $soap, head => $ahdr }, $sections )
+sub do_getinfo {
+    my $opt = ( ref( $_[0] ) eq "HASH" ) ? shift : {};
+    my ($sections) = @_;
+
+    my $attr = { xmlns => NS_ZACCOUNT };
+    $attr->{sections} = $sections if $sections;
+
+    # ZCS http/soap proxy chokes on requests if SOAP::Data uses xsi:nil="true"
+    my $body = [ [] ];
+
+    return call_soap(
+        %$opt,
+        req  => "GetInfoRequest",
+        attr => $attr,
+        body => $body
+    );
+}
+
+sub call_soap {
+    my %opt = @_;
+
+    # die on errors by default
+    unless ( exists $opt{RaiseError} or exists $opt{PrintError} ) {
+        $opt{RaiseError} = 1;
+    }
+
+    my $handler =
+        $opt{RaiseError} ? sub { die(@_) }
+      : $opt{PrintError} ? sub { warn(@_) }
+      :                    undef;
+
+    my $soap = $opt{soap};
+    my $req  = SOAP::Data->name( $opt{req} );
+    $req->attr( $opt{attr} ) if $opt{attr};
+
+    my $resp;
+    my ( $try, $maxtry, $sleep ) = ( 0, 3, 0 );
+    while ( !$resp && $try++ < $maxtry ) {
+		$soap->{_transport}->{_proxy}->{ssl_opts}->{verify_hostname} = 0;
```

Comment:
nit: tabs to spaces (and align properly)
fix: documenting that we're not ensuring that servers that have a valid certificate matching the expected hostname

Reply:
nit - done
fix- removed bypassing ssls_opts, so that ssl check will be done.